lib/rules/no-global-setup.js

```diff
@@ -10,24 +10,37 @@ var setupRegexp = /^(before|after)(Each|All)$/
 
 module.exports = function (context) {
   var suiteDepth = 0
+  var hasGlobalSetup = false
+  var hasRootSuite = false
+  var globalSetupNames = []
   return {
     CallExpression: function (node) {
       if (suiteRegexp.test(node.callee.name)) {
+        hasRootSuite = true
         suiteDepth++
       } else if (setupRegexp.test(node.callee.name) && suiteDepth === 0) {
-        context.report({
-          data: {
-            name: node.callee.name
-          },
-          message: 'Do not use `{{name}}` outside a `describe`.',
-          node
-        })
+        hasGlobalSetup = true
+        globalSetupNames.push(node.callee.name)
       }
     },
     'CallExpression:exit': function (node) {
       if (suiteRegexp.test(node.callee.name)) {
         suiteDepth--
       }
+    },
+    'Program:exit': function () {
+      if (hasGlobalSetup && hasRootSuite) {
+        context.report({
+          data: {
+            names: globalSetupNames.join(', ')
+          },
+          message: 'Do not use `{{names}}` outside a `describe` except for in global helpers.',
+          loc: {
```

Comment:
Not terribly proud of this... Two thoughts:

1. Keep it because it's appropriate to the issue being in the global context of the unit in question.
1. Explicitly bind the location of the first incident in the report.